tmt/export.py

```diff
@@ -0,0 +1,189 @@
+        fmf_case_attrs['tag'].append('fmf-export')
+        tag_list = [nitrate.Tag(tag) for tag in fmf_case_attrs['tag']]
+        nitrate_case.tags.add(tag_list)
+        echo(style('tags: ', fg='green') +
+             ' '.join(set(fmf_case_attrs['tag'])))
+
+        # Default tester
+        try:
+            email_address = email.utils.parseaddr(fmf_case_attrs['contact'])[1]
+            # TODO handle nitrate user not existing and other stuff
+            nitrate_case.tester = nitrate.User(email_address)
+            echo(style('default tester: ', fg='green') + email_address)
+        except KeyError:
+            # Defaults to author of the test case
+            pass
+
+        # Duration
+        try:
+            nitrate_case.time = fmf_case_attrs['duration']
+            echo(style('estimated time: ', fg='green') +
+                 fmf_case_attrs['duration'])
+        except KeyError:
+            # Defaults to 5 minutes
+            nitrate_case.time = '5m'
+            echo(style('estimated time: ', fg='green') + '5m')
+
+        # Status
+        try:
+            if fmf_case_attrs['enabled']:
+                nitrate_case.status = nitrate.CaseStatus('CONFIRMED')
+                echo(style('status: ', fg='green') + 'CONFIRMED')
+            else:
+                nitrate_case.status = nitrate.CaseStatus('DISABLED')
+                echo(style('status: ', fg='green') + 'DISABLED')
+        except KeyError:
+            # Defaults to enabled
+            nitrate_case.status = nitrate.CaseStatus('CONFIRMED')
+            echo(style('status: ', fg='green') + 'CONFIRMED')
+
+        # Environment
+        try:
+            env = ' '.join(tmt.utils.dict_to_shell(
+                fmf_case_attrs['environment']))
+            nitrate_case.arguments = env
+            echo(style('arguments: ', fg='green') + env)
+        except KeyError:
+            # FIXME unable to set empty arguments, possibly error in xmlrpc, BZ#1805687
+            nitrate_case.arguments = ' '
+            echo(style('arguments: ', fg='green') + "' '")
+
+        struct_field = tmt.utils.StructuredField(nitrate_case.notes)
+        echo(style('Structured Field: ', fg='blue'))
+
+        # Mapping of structured field sections to fmf case attributes
+        section_to_attr = {'relevancy': 'relevancy', 'description': 'summary',
+                           'purpose-file': 'description', 'hardware': 'extra-hardware', 'pepa': 'extra-pepa'}
+
+        for section, attribute in section_to_attr.items():
+            try:
+                struct_field.set(section, fmf_case_attrs[attribute])
+                echo(style(section + ': ', fg='green') +
+                     fmf_case_attrs[attribute].strip())
+            except KeyError:
+                pass
+
+        # fmf identifer
+        fmf_id = create_fmf_id(path=os.path.abspath('.'),
+                               name=fmf_case.name)[0]
+
+        struct_field.set('fmf', pprint.pformat(fmf_id))
```

Comment:
The identifier should not be stored as `dict` under the structured field:

    [purpose-file]
    Just run 'tmt --help' to make sure the binary is sane.
    This is really that simple. Nothing more here. Really.

    [fmf]
    {'name': '/',
    'path': '/home/psss/git/tmt/examples/convert',
    'ref': 'test-export',
    'url': 'github.com:psss/tmt.git'}

    [structured-field-end]

It should be stored there as a `yaml` dictionary. See fmf identifier [examples](https://fmf.readthedocs.io/en/latest/concept.html#identifiers).

Reply:
Improved by b80f188

Reply:
Looks good, thanks.